discord_slash/client.py

```diff
@@ -847,6 +879,230 @@ def wrapper(cmd):
 
         return wrapper
 
+    def add_component_callback(
+        self,
+        callback: typing.Coroutine,
+        *,
+        message_id: int = None,
+        message_ids: typing.List[int] = None,
+        custom_id: str = None,
+        custom_ids: typing.List[str] = None,
+        use_callback_name=True,
+        component_type: int = None,
+    ):
+        """
+        Adds a coroutine callback to a component.
+        Callback can be made to only accept component interactions from a specific messages
+        and/or custom_ids of components.
+
+        .. note::
+            ``message_id`` and ``message_ids`` cannot be used at the same time. The same applies to ``custom_id`` and ``custom_ids``.
+
+        :param callback: The coroutine to be called when the component is interacted with. Must accept a single argument with the type :class:`.context.ComponentContext`.
+        :type callback: Coroutine
+        :param message_id: If specified, only interactions from the message given will be accepted.
+        :type message_id: Optional[int]
+        :param message_ids: Similar to ``message_id``, but accepts a list of message IDs instead. Empty list will mean that no interactions are accepted.
+        :type message_ids: Optional[List[int]]
+        :param custom_id: The ``custom_id`` of the component. Defaults to the name of ``callback`` if ``use_callback_name=True``.
+        :type custom_id: Optional[str]
+        :param custom_ids: Similar to ``custom_ids``, but accepts a list of custom IDs instead. Empty list will mean that no interactions are accepted.
+        :type custom_ids: Optional[List[str]]
+        :param use_callback_name: Whether the ``custom_id`` defaults to the name of ``callback`` if unspecified. If ``False``, either `message_ids`` (``message_id``) or ``custom_ids`` (``custom_id``) must be specified.
+        :type use_callback_name: bool
+        :param component_type: The type of the component to avoid collisions with other component types. See :class:`.model.ComponentType`.
+        :type component_type: Optional[int]
+        :raises: .error.DuplicateCustomID, .error.IncorrectFormat
+        """
+        if message_id and message_ids is not None:
+            raise error.IncorrectFormat("You cannot use both `message_id` and `message_ids`!")
+
+        if custom_id and custom_ids is not None:
+            raise error.IncorrectFormat("You cannot use both `custom_id` and `custom_ids`!")
```

Comment:
Would it not be more user friendly to have just `custom_ids` and have it handle `str` and `list[str]`

Reply:
Maybe, I just was trying to be consistent with stuff like `embed` and `embeds`

Reply:
Changed all signatures of callback adding functions to include suggested changes. They also now accept whole message objects and component dicts (or lists of those too)